packages/sessions-sdk-react/src/components/deposit-page.tsx

```diff
@@ -21,71 +21,127 @@ import { StateType, useData } from "../hooks/use-data.js";
+  </div>
+);
+
+const DepositPageContents = (props: Omit<Props, "onPressBack">) => {
   const { getSessionContext, network } = useSessionContext();
-  const getSolanaBalance = useCallback(async (): Promise<SolanaBalance> => {
+  const getSolanaBalance = useCallback(async (): Promise<SolanaBalances> => {
     const { getSolanaConnection } = await getSessionContext();
     const connection = await getSolanaConnection();
-    try {
-      const result = await connection.getTokenAccountBalance(
-        getAssociatedTokenAddressSync(
-          USDC.chains[network].solana.mint,
-          props.sessionState.walletPublicKey,
-        ),
-      );
-      return result.value;
-    } catch (error: unknown) {
-      if (
-        error instanceof SolanaJSONRPCError &&
-        error.message === NO_ACCOUNT_MESSAGE
-      ) {
-        return {
-          amount: "0",
-          uiAmountString: "0",
-        };
-      } else {
-        throw error;
-      }
-    }
+    const [usdc, sol] = await Promise.all([
+      getSolanaUsdcBalance(
+        connection,
+        network,
+        props.sessionState.walletPublicKey,
+      ),
+      connection.getBalance(props.sessionState.walletPublicKey),
+    ]);
+    return { usdc, sol };
   }, [getSessionContext, props.sessionState.walletPublicKey, network]);
-  const balance = useData(
-    ["solanaUsdcBalance", network, props.sessionState.walletPublicKey],
+  const solanaBalances = useData(
+    ["solanaBalances", network, props.sessionState.walletPublicKey],
     getSolanaBalance,
     {},
   );
 
-  return (
-    <div className={styles.depositPage ?? ""}>
-      <Button
-        onPress={onPressBack}
-        variant="outline"
-        className={styles.backButton ?? ""}
-      >
-        Back
-      </Button>
-      <DepositForm
-        {...props}
-        {...(balance.type === StateType.Loaded
-          ? {
-              amountAvailable: balance.data,
-              mutateAmountAvailable: balance.mutate,
-            }
-          : { isLoading: true })}
-      />
-    </div>
-  );
+  switch (solanaBalances.type) {
+    case StateType.Error: {
+      return (
+        <FetchError
+          className={styles.fetchError}
+          headline="Failed to load Solana account balances"
+          error={solanaBalances.error}
+          reset={solanaBalances.reset}
+        />
+      );
+    }
+    case StateType.Loaded: {
+      if (solanaBalances.data.sol === 0) {
+        return (
+          <FetchError
+            className={styles.fetchError}
+            headline="No SOL in your Solana wallet"
+            error="You must have SOL in your Solana wallet to pay gas on Solana to transfer to Fogo."
+          />
+        );
+      } else if (solanaBalances.data.usdc.amount === "0") {
+        return (
+          <FetchError
+            className={styles.fetchError}
+            headline="No USDC in your Solana wallet"
+            error="You have no USDC on Solana to transfer to Fogo."
```

Comment:
should this one be first?
like if the wallet is completely empty seems like the most logical error to raise is "you don't have USDC"